bin/tx-prover/src/server/mod.rs

```diff
@@ -0,0 +1,88 @@
+use generated::api::{api_server, ProveTransactionRequest, ProveTransactionResponse};
+use miden_objects::transaction::{ProvenTransaction, TransactionWitness};
+use miden_tx::{
+    utils::{Deserializable, Serializable},
+    LocalTransactionProver, TransactionProver, TransactionProverError,
+};
+use tokio::{net::TcpListener, sync::Mutex};
+use tonic::{Request, Response, Status};
+use tracing::info;
+use winter_maybe_async::maybe_await;
+
+pub mod generated;
+
+pub struct Rpc {
+    pub api_service: api_server::ApiServer<RpcApi>,
+    pub listener: TcpListener,
+}
+
+impl Rpc {
+    pub fn new(listener: TcpListener) -> Self {
+        let api_service = api_server::ApiServer::new(RpcApi::default());
+        Self { listener, api_service }
+    }
+}
+
+#[derive(Default)]
+pub struct RpcApi {
+    prover: Mutex<LocalTransactionProver>,
+}
+
+// We need to implement Send and Sync for the generated code to be able to use the prover in the
+// shared context.
+unsafe impl Send for RpcApi {}
+unsafe impl Sync for RpcApi {}
```

Comment:
Is this still required (even though we use `Mutex`)?

Reply:
It is required because `Rc<TransactionMastStore>` is not thread safe and does not implement `Send`. We can use `Arc` instead of `Rc`, or add the implementation of `Send` to the `LocalTransactionProver` though I consider that implementing it only for RpcApi is the "cleanest" option.

Reply:
Let's use `Arc` instead of `Rc`.